history.go

```diff
@@ -54,11 +68,15 @@ func (h *History) UpdateHistory(policies Policies, m *MerkleTree, nextEp Epoch)
 		}
 		h.loadedEpochs = append(h.loadedEpochs[:0], h.loadedEpochs[n:]...)
 	}
-	h.snapshots[nextEp] = nextStr
-	h.loadedEpochs = append(h.loadedEpochs, nextEp)
+
+	h.updateInternal(policies, nextEp)
 	return nil
 }
 
+func (h *History) Set(key string, value []byte) error {
```

Comment:
> What do you mean "the server level"? Is it our libconiks-server library? 

Yes, I mean the libconiks-server library. 

> I think it's fine to have these wrappers in this library. We can think these wrappers as APIs of the library, and in the libconiks-server library, we should have something like ``directory.LookUp`` and ``directory.Set()``. Is that what you mean? 

I don't really know if ``History`` makes the most sense as the interface, though. I've been thinking of ``History`` more as the structure that gives developers access to older versions of the Merkle tree. Maybe this is coming back to an earlier discussion we had about the naming of ``History``. I don't see the problem in exposing the tree setter and getter methods to the server directly since we're exposing many other aspects of the tree anyway (e.g. the ``MerkleNode`` from the auth path).